lib/_tls_common.js

```diff
@@ -132,6 +134,52 @@ exports.createSecureContext = function createSecureContext(options, context) {
     }
   }
 
+  if (options.pskCallback && c.context.enablePskCallback) {
+    c.context.onpskexchange = (identity, maxPskLen, maxIdentLen) => {
+      let ret = options.pskCallback(identity);
+      if (typeof ret !== 'object') {
+        ret = undefined;
+      }
+
+      if (ret) {
+        ret = { psk: ret.psk, identity: ret.identity };
+
+        if (typeof ret.psk === 'string') {
+          ret.psk = Buffer.from(ret.psk);
+        } else if (!Buffer.isBuffer(ret.psk)) {
+          throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'psk',
+                                     ['string', 'buffer']);
+        }
+        if (ret.psk.length > maxPskLen) {
+          throw new errors.RangeError(
```

Comment:
Also, how is the formatting on these lines? It complies with eslint, but it's ugly as sin.